### PR TITLE
Update for IoT OT Threat Monitoring Solution Package

### PR DESCRIPTION
**Pull request by amitcohen25**
Required items, please complete
   
   Change(s):
   - 3 new playbooks: **AD4IOT-CVEAutoWorkflow, AD4IOT-SendEmailToIoTOwner, AD4IOT-AutoTriage**
   - Update for the existing workbook: **IoTOTThreatMonitoringwithDefenderforIoT**
   - Please remove from the solution package the playbook called: **AD4IoT-MailByProductionLine**
   - Please remove from the solution packge the playbook called: **Get-AD4IoTDeviceCVEs - Incident**
   - Please change the solution package name to: **Microsoft Defender for IoT** instead of IoT OT Threat Monitoring with Defender for IoT
   - Please change the workbook name to **Microsoft Defender for IoT** from IoTOTThreatMonitoringwithDefenderforIoT
   - Please add Defender for IoT Data Connector to the solution package
   - Please convert the solution package to V2

   Reason for Change(s):
   - Updates following customers feedbacks

   Testing Completed:
   - Yes

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes

# Guidance <- remove section before submitting
-----------------------------------------------------------------------------------------------------------
## Before submitting this PR please ensure that you have read the following sections and filled out the changes, reason for change and testing complete sections:

Thank you for your contribution to the Microsoft Sentinel Github repo.

> Details of the code changes in your submitted PR.  Providing descriptions for pull requests ensures there is context to changes being made and greatly enhances the code review process.  Providing associated Issues that this resolves also easily connects the reason.
   
   Change(s):
   - Updated syntax for XYZ.yaml

   Reason for Change(s):
   - New schema used for XYZ.yaml
   - Resolves ISSUE #1234

   Version updated:
   - Yes
   - Detections/Analytic Rule templates are required to have the version updated

> The code should have been tested in a Microsoft Sentinel environment that does not have any custom parsers, functions or tables, so that you validate no incorrect syntax and execution functions properly.  If your submission requires a custom parser or function, it must be submitted with the PR.

   Testing Completed:
   - Yes/No/Need Help

_Note: If updating a detection, you must update the version field._

> Before the submission has been made, please look at running the KQL and Yaml Validation Checks locally.
> https://github.com/Azure/Azure-Sentinel#run-kql-validation-locally

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes/No/Need Help
   
   _Note: Let us know if you have tried fixing the validation error and need help._

> References: 
> - [Guidance for Detection checks](https://github.com/Azure/Azure-Sentinel#pull-request-detection-template-structure-validation-check)
> - [General contribution guidance](https://github.com/Azure/Azure-Sentinel/wiki#what-can-you-contribute-and-how-can-you-create-contributions)
> - [PR validation troubleshooting](https://github.com/Azure/Azure-Sentinel#pull-request)

-----------------------------------------------------------------------------------------------------------

### CODE REVIEW

**Review by aprakash13 — COMMENTED** [review]
No major change from an analytic rule perspective. Please feel free to merge if other components look good and the PR has passed all the validation checks.

**Comment by v-mchatla** [comment]
Hi @amitcohen25
Can you please fix the failed checks.
Thanks

**Review by NikTripathi — CHANGES_REQUESTED** [review]
Hi @amitcohen25 In workbook for few queries I am getting below output.
image
We don't have any option for selecting those values. Is it expected?

File: Solutions/IoTOTThreatMonitoringwithDefenderforIoT/Workbooks/IoTOTThreatMonitoringwithDefenderforIoT.json
Comment (NikTripathi) [inline Solutions/IoTOTThreatMonitoringwithDefenderforIoT/Workbooks/IoTOTThreatMonitoringwithDefenderforIoT.json:100]: Hi @amitcohen25 Please remove these hardcoded values. Because of these validations are failing.
Thanks

**Review by NikTripathi — CHANGES_REQUESTED** [review]

File: Solutions/IoTOTThreatMonitoringwithDefenderforIoT/Workbooks/IoTOTThreatMonitoringwithDefenderforIoT.json
Comment (amitcohen25) [reply Solutions/IoTOTThreatMonitoringwithDefenderforIoT/Workbooks/IoTOTThreatMonitoringwithDefenderforIoT.json]: Hi @NikTripathi , thanks for highlighting this. I fixed it

Comment (amitcohen25) [reply Solutions/IoTOTThreatMonitoringwithDefenderforIoT/Workbooks/IoTOTThreatMonitoringwithDefenderforIoT.json]: @NikTripathi The output you shared (missing parameters) is fine. You should first select a record from the table and then it will work

**Review by NikTripathi — APPROVED** [review]
Workbook looks good. Can approve once playbooks are approved.
Thanks

**Comment by v-mchatla** [comment]
Hi @rushriva 
Can you please review playbooks and provide your feedback.
Thanks

**Comment by NikTripathi** [comment]
@amitcohen25 thanks for clarification. Once playbooks are approved we will go ahead with rest of the steps.

File: Solutions/IoTOTThreatMonitoringwithDefenderforIoT/Analytic Rules/IoTUnauthorizedPLCModifications.yaml
Comment (shainw) [inline Solutions/IoTOTThreatMonitoringwithDefenderforIoT/Analytic Rules/IoTUnauthorizedPLCModifications.yaml:82]: Why are we using `contains` instead of `has` in both line 81 and 82?  Also, I would suggest for lines 22 thru 82 you use a [dynamic](https://docs.microsoft.com/en-us/azure/data-explorer/kusto/query/scalar-data-types/dynamic) list for the AlertName matching that is currently an embedded list in the query and then use [`has_any`](https://docs.microsoft.com/en-us/azure/data-explorer/kusto/query/has-anyoperator#use-has_any-operator-with-a-dynamic-array) against that dynamic list.

**Review by shainw — CHANGES_REQUESTED** [review]
Version needs to be updated, not sure why the validation is not picking it up.

File: Solutions/IoTOTThreatMonitoringwithDefenderforIoT/Analytic Rules/IoTUnauthorizedPLCModifications.yaml
Comment (amitcohen25) [reply Solutions/IoTOTThreatMonitoringwithDefenderforIoT/Analytic Rules/IoTUnauthorizedPLCModifications.yaml]: @shainw Thanks! Changes done. could you please review?

Comment (amitcohen25) [reply Solutions/IoTOTThreatMonitoringwithDefenderforIoT/Analytic Rules/IoTUnauthorizedPLCModifications.yaml]: @shainw what's the next step?

**Comment by rushriva** [comment]
For Playbooks- Each folder must contain azuredeploy.json file, readme.md and a folder with name "images" if readme is referencing any local images (optional). Please add readme file for playbooks.

**Comment by rushriva** [comment]
Playbook changes are approved.

**Comment by amitcohen25** [comment]
@NikTripathi I can see the playbook changes are approved. are we ready to package and publish the changes?
Also, could you pelase supprt those requests as well as part of this change:
1. Please remove from the solution package the playbook called: AD4IoT-MailByProductionLine
2. Please change the solution package name to: Defender for IoT instead of IoT OT Threat Monitoring with Defender for IoT
3. Please change the workbook name to Defender for IoT from IoTOTThreatMonitoringwithDefenderforIoT
4. Please convert the solution package to V2 so that users will be able to manage their package.

File: Solutions/IoTOTThreatMonitoringwithDefenderforIoT/Analytic Rules/IoTUnauthorizedPLCModifications.yaml
Comment (shainw) [inline Solutions/IoTOTThreatMonitoringwithDefenderforIoT/Analytic Rules/IoTUnauthorizedPLCModifications.yaml:22]: Please don't do it this way, it will add this to every row. Simply use a let statement at the top of the query:

`let alertList = dynamic(["Unpermitted Usage of Internal Indication (IIN)", "Modbus Address Range Violation", etc.]);`

**Review by shainw — CHANGES_REQUESTED** [review]
Please change to using a let statement to make a single list versus writing the list into every row in the results.

File: Solutions/IoTOTThreatMonitoringwithDefenderforIoT/Analytic Rules/IoTUnauthorizedPLCModifications.yaml
Comment (amitcohen25) [reply Solutions/IoTOTThreatMonitoringwithDefenderforIoT/Analytic Rules/IoTUnauthorizedPLCModifications.yaml]: @shainw done

**Comment by NikTripathi** [comment]
Hi @amitcohen25 ,

Validations are failing because of the Analytic rule.

Error Message:
   Expected object to be <null> because In template IoTUnauthorizedPLCModifications.yaml there was an error while parsing: (Line: 22, Col: 1, Idx: 3301) - (Line: 22, Col: 3, Idx: 3303): While scanning a block scalar, did not find expected comment or line break., but found YamlDotNet.Core.SyntaxErrorException with message "(Line: 22, Col: 1, Idx: 3301) - (Line: 22, Col: 3, Idx: 3303): While scanning a block scalar, did not find expected comment or line break."
   
   
   Incorrect yaml file. File path: Solutions/IoTOTThreatMonitoringwithDefenderforIoT/Analytic Rules/IoTUnauthorizedPLCModifications.yaml. Error message: can not read a block mapping entry; a multiline key may not be an implicit key at line 50, column 15:
    entityMappings:


Can you please fix this?

**Review by NikTripathi — APPROVED** [review]

**Review by shainw — APPROVED** [review]

**Pull request merged**
